### ISSUE TITLE
[Request]

### ISSUE THREAD

**Issue by yet-another-average-joe** [OctoPrint/OctoPrint#3560]
**Is your feature request related to a problem? Please describe.**

When a plugin edits config.yaml, a backup file is created : config.backup. But one backup is not enough in some circumstances

**Describe the solution you'd like**

Maintaining a few backup files :  config.backup01,  config.backup02, and so on ; this very common in the software industry ; for example, Eagle CAD does it.

**Describe alternatives you've considered**

Doing backups by hand. Using a SSH console, or doing it from a computer after .octoprint is shared with samba.

**Additional context**

none

**Comment by tedder** [comment]
Please edit the subject to describe your feature request.

If you are concerned about backups, why not use the backup UI for the whole system?

![Screenshot from 2020-05-09 11-38-15](https://user-images.githubusercontent.com/97914/81482145-bcd98180-91e9-11ea-92b1-6f60d56d9c41.png)

**Comment by yet-another-average-joe** [comment]
I just learn now about the backup/restore ! :))) I never payed attention, probably because I've always been used to image my computers system disk : with OctoPi, I just put the micro SD card in another Pi, shrink partitions, and image the whole card on a regular basis ; or save individual files on a PC through network shares. I've been doing this for the last 2 years on OctoPi !

Now, I will also use this feature, it's less time consuming !

Long story short, this FR was an idea I had after I read this : https://github.com/Salandora/octoprint-customControl/issues/32#issuecomment-626177415 ; it's only about some basic auto backup or undo queue, just in case a plugin does something wrong with config.yaml. (never had such an issue, but some did)

**Comment by tedder** [comment]
Cool. Yeah, I could see the desire for an autobackup type feature, but it seems like a lot of work for little gain.
